interfaces/system_key.go

```diff
@@ -80,39 +119,86 @@ func generateSystemKey() *systemKey {
 		logger.Noticef("cannot determine root filesystem on overlay in generateSystemKey: %v", err)
 	}
 
-	// Add the current Core path, we need this because we call helpers
-	// like snap-confine from core that will need an updated profile
-	// if it changes
-	//
-	// FIXME: what about core18? the snapd snap?
-	sk.Core, _ = os.Readlink(filepath.Join(dirs.SnapMountDir, "core/current"))
-
 	// Add seccomp-features
 	sk.SecCompActions = release.SecCompActions()
 
-	return &sk
+	return sk
 }
 
-// SystemKey outputs a string that identifies what security profiles
-// environment this snapd is using. Security profiles that were generated
-// with a different Systemkey should be re-generated.
-func SystemKey() string {
-	sk := generateSystemKey()
+// WriteSystemKey will write the current system-key to disk
+func WriteSystemKey() error {
+	sks, err := json.Marshal(generateSystemKey())
+	if err != nil {
+		panic(err)
```

Comment:
Why not return the error if we have the chance?

Reply:
+1, let's please not panic.

Reply:
Yes, good point!